src/Config/ConfigFileWatcher.cs

```diff
@@ -31,14 +38,19 @@ private void OnConfigFileChange(object sender, FileSystemEventArgs e)
     {
         try
         {
-            if (_configProvider is null)
+            if (_configLoader is null)
+            {
+                throw new ArgumentNullException("ConfigLoader can not be null.");
+            }
+
+            if (_configLoader.RuntimeConfig is null)
             {
-                throw new ArgumentNullException("_configProvider can not be null.");
+                throw new ArgumentNullException("RuntimeConfig can not be null.");
             }
 
-            if (!_configProvider.IsLateConfigured && _configProvider.GetConfig().IsDevelopmentMode())
+            if (_configLoader.RuntimeConfig.IsDevelopmentMode())
             {
-                _configProvider.HotReloadConfig();
+                _configLoader.HotReloadConfig(_configLoader.RuntimeConfig.DefaultDataSourceName);
```

Comment:
may be helpful to print/log that hot reloading was skipped  because config definded production mode.